src/raytracer2018.nim

```diff
@@ -1401,6 +1401,60 @@ proc computeMagnetTransmission(
     #echo "axion mass in [eV] ", mAxion, " effective photon mass in [eV] ", effPhotonMass
     result = cos(ya) * probConversionMagnetGas * absorbtionXrays # for setup with gas
 
+proc lineIntersectsOpaqueTelescopeStructures(
+  expSetup: ExperimentSetup,
+  testVector, vectorEntranceXRTCircular, pointExitCB, pointEntranceXRT: Vec3[float],
+  onlyInnerPart = false
+     ): bool =
+
+  ## XXX: `onlyInnerPart` is a hack? for the case where we only want to enter the
+  ## first branch of the `if vectorEntrance...` for `esBabyIAXO`.
+  ##
+  ## XXX: Also the expSetup should become a `telescopeKind` case, no?
+  ##
+  ## XXX: can we please clean up the esBabyIAXO branch? Remove the commented code?
+
+  ## there is a 2mm wide graphite block between each glass mirror, to seperate them
+  ## in the middle of the X-ray telescope. Return if hit
+  ## BabyIAXO return if X-ray its the spider structure
+  case expSetup.kind
+  of esCAST:
+    ## graphite blockers of the LLNL optics
+    if pointEntranceXRT[1] <= 1.0 and pointEntranceXRT[1] >= -1.0: return
+  of esBabyIAXO:
+    ## here we have a spider structure for the XMM telescope:
+    ## XXX: the checks for the `vectorEntranceXRTCircular` stuff is extremely confusing!
+    if onlyInnerPart or vectorEntranceXRTCircular[0] <= 64.7: #and vectorEntranceXRTCircular[0] > expSetup.holeInOptics.float * expSetup.numberOfHoles.float: #doesnt really matter because there are no mirrors in the middle and these axions dont reach the window anyways
+      let nHoles = expSetup.telescope.numberOfHoles
+      for l in -(nHoles - ceil(nHoles.float / 2.0).int) .. (nHoles - ceil(nHoles.float / 2.0).int):
+        var centerHole = testVector
+        if l != 0:
+          if abs(l) mod 2 == 0:
+            centerHole[1] += 2.0 * l.float * expSetup.telescope.holeInOptics.float
+          else:
+            centerHole[0] += 2.0 * (l + (l / abs(l))).float * expSetup.telescope.holeInOptics.float
+          #if abs(l) < 3: typeHole = expSetup.holetype
+          #else: typeHole = "square"
+        #else: typeHole = "square"
+        if lineIntersectsObject(expSetup.telescope.holeType, pointExitCB, pointEntranceXRT,
+                                centerHole, expSetup.telescope.holeInOptics):
+          result = true
+          break
+        else:
+          result = false
+    # hole max 20.9 mm doesnt really matter because there are no mirrors in the
+    # middle and these axions dont reach the window anyways
+    elif vectorEntranceXRTCircular[0] < 151.6 and vectorEntranceXRTCircular[0] > (151.6 - 20.9):
+      result = false
+    elif vectorEntranceXRTCircular[0] > 64.7:
+      ## XXX: why 16? strips?
```

Comment:
yes

Reply:
So that should eventually also become part of the telescope type? Or is the spider structure so specific to that telescope that this doesn't make sense?

Reply:
It's too specific I think. Some telescopes have it and some don't